src/db/postgres.rs

```diff
@@ -39,4 +44,29 @@ impl Store for PostgresStore {
 
         Ok(row.0 == 0)
     }
+
+    async fn check_password<'a>(
+        &self,
+        user_id: &'a UserIdentifier,
+        password: &str,
```

Comment:
This seems dangerous. However we deal with passwords, it should be storage independent. At best, it seems like this should return salt+hash+encoding and leave the verification to a common util. I'd normally say "eh, whatever, don't over-engineer right now" but I would rather just limit auth to OTP until we can handle passwords correctly.